vms/proposervm/block.go

```diff
@@ -163,9 +194,18 @@ func (p *postForkCommonComponents) Verify(
 	}
 
 	var contextPChainHeight uint64
-	if p.vm.Upgrades.IsEtnaActivated(childTimestamp) {
+	switch {
+	case p.vm.Upgrades.IsGraniteActivated(childTimestamp):
```

Comment:
Unrelated to this PR, but, food for thought for whoever is reading this review - shouldn't we unify the code here and the code when we verify a block? It' generally bad practice to have a version of the code that verifies input and a different version that builds the same output. Ideally we'll have a single function that computes the output properly and then the verification function will just compute it exactly as the block build process and then just compare. 

This switch of IsGrantite?IsEtna?IsSomethingElse? should ideally be extracted to a separate function. 

Thoughts?

Reply:
Completely agree with the premise of this comment, and that it's not for this PR either way.

However, for block building specifically this seems difficult/impossible because there are many different valid ways to build the next block, so we can't just simply build locally and compare to what another build proposed.